be/src/olap/rowset/segment_v2/binary_prefix_page.cpp

```diff
@@ -0,0 +1,250 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/rowset/segment_v2/binary_prefix_page.h"
+
+#include <map>
+#include <vector>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <string>
+
+#include "common/logging.h"
+#include "gutil/strings/substitute.h"
+#include "runtime/mem_pool.h"
+#include "util/coding.h"
+#include "util/faststring.h"
+#include "util/slice.h"
+
+namespace doris {
+namespace segment_v2 {
+
+using strings::Substitute;
+
+Status BinaryPrefixPageBuilder::add(const uint8_t* vals, size_t* add_count) {
+    DCHECK(!_finished);
+    if (*add_count == 0) {
+        return Status::OK();
+    }
+
+    const Slice* src = reinterpret_cast<const Slice*>(vals);
+    if (_count == 0) {
+        _first_entry.assign_copy(reinterpret_cast<const uint8_t*>(src->get_data()), src->get_size());
+    }
+
+    int i = 0;
+    for (; i < *add_count; ++i, ++src) {
+        if (is_page_full()) {
+            break;
+        }
+        char* entry = src->data;
+        size_t entry_len = src->size;
+        int old_size = _buffer.size();
+
+        int share_len;
+        if (0 == _count % RESTART_POINT_INTERVAL) {
```

Comment:
```suggestion
        if (_count % RESTART_POINT_INTERVAL == 0) {
```
Is there any reason for choosing fixed interval restart point? If there is, better to write it down as comment

Reply:
It's a empirical value, Kudu and LevelDB use 16. I has written the RESTART_POINT_INTERVAL to page footer.